rfd/0086-stream-events-api.md

```diff
@@ -0,0 +1,234 @@
+---
+authors: Vitor Enes (vitor@goteleport.com)
+state: draft
+---
+
+# RFD 86 - Stream Events API
+
+## Required Approvals
+
+* Engineering: @r0mant && @jimbishopp
+
+## Table of Contents
+
+* [What](#what)
+* [Why](#why)
+  * [Goals](#goals)
+  * [Non\-Goals](#non-goals)
+* [Details](#details)
+  * [`Client.StreamEvents` API](#clientstreamevents-api)
+  * [`StreamEvents` RPC](#streamevents-rpc)
+  * [`IAuditLog.StreamEvents` API](#iauditlogstreamevents-api)
+  * [`dynamoevents.Log.StreamEvents` API](#dynamoeventslogstreamevents-api)
+	* [DynamoDB stream cursor](#dynamodb-stream-cursor)
+	* [`lib/backend/dynamo/shards.go`](#libbackenddynamoshardsgo)
+* [Concerns and open questions](#concerns-and-open-questions)
+
+## What
+
+Currently streaming of audit events has to be implemented on top of [`SearchEvents`], as [done by the event-handler] plugin.
```

Comment:
What about `StreamSessionEvents`? I think it only works for session recording events and not arbitrary audit events, but it still probably makes sense to discuss here and see if there's any overlap.

Reply:
Yes, `StreamingSessionEvents` only works for session recordings. I'll add a note about it.

Reply:
Done in 2e9189beda9447b6d64c935ab32e3a8359e50da4.